src/SilverStripe/LocalGit/GitProcess.php

```diff
@@ -42,23 +55,38 @@ public function getIdentityFile() {
 
 	/**
 	 * @param string $file Path to private key
-	 * @throws \RuntimeException
 	 */
 	public function setIdentityFile($file) {
 		if(is_file($file) && is_readable($file)) {
 			$this->identityFile = $file;
 		} else {
-			throw new \RuntimeException(sprintf('%s does not exist or is not readable.', $file));
+			throw new Exception(sprintf('%s does not exist or is not readable.', $file));
 		}
 	}
 
 	/**
+	 * @return string|null
+	 */
+	public function getKnownHostsFile()
+	{
+		return $this->knownHostsFile;
+	}
+
+	/**
+	 * @param string $file Path to known hosts file. Must be read/writable.
+	 */
+	public function setKnownHostsFile($file) {
+		$this->knownHostsFile = $file;
+	}
+
+	/**
 	 * @param null $callback
 	 * @return null|int null or 0 if everything went fine, or an error code
 	 */
 	public function run($callback = null) {
 		$this->setCommandLine(
 			'env IDENT_KEY=' . escapeshellarg($this->getIdentityFile())
```

Comment:
Can we use something like `http://api.symfony.com/3.0/Symfony/Component/Process/ProcessBuilder.html#method_addEnvironmentVariables` instead of passing this as a command line thing for a bit more readable code?

Reply:
Done - have used ->setEnv instead of ProcessBuilder because we `extend Process`